syntax = "proto3";

service Scheduler {
  rpc GetTrainStatus (TrainStatusRequest) returns (TrainStatusResponse) {}
  rpc UpdateTrainStatus (TrainUpdateRequest) returns (TrainUpdateResponse) {}
  rpc GetOtherTrainStatus (OtherTrainStatusRequest) returns (TrainStatusResponse) {}

}

message OtherTrainStatusRequest {
  int32 requesting_train_id = 1;
  int32 other_train_id = 2;
}

message TrainStatusRequest {
  int32 train_id = 1;
}

message TrainStatusResponse {
  int32 train_id = 1;
  double location = 2;
  double speed = 3;
  double distance_to_stop = 4;
}

message TrainUpdateRequest {
  int32 train_id = 1;
  double location = 2;
  double speed = 3;
}

message TrainUpdateResponse {
  bool success = 1;
}
